larq/models.py

```diff
@@ -47,91 +67,316 @@ def _get_output_shape(layer):
         return "?"
 
 
-def _compute_memory(layer_stat):
-    mem_in_bits = sum(precision * number for precision, number in layer_stat.items())
-    return _bit_to_kB(mem_in_bits)
+class _parameterProfile:
+    bitwidth = None
 
+    def __init__(self, parameter, bitwidth=32):
+        self._parameter = parameter
+        self.bitwidth = bitwidth
 
-def _parse_params(layer):
-    if hasattr(layer, "quantized_latent_weights"):
-        params = {}
-        for quantizer, weight in zip(layer.quantizers, layer.quantized_latent_weights):
-            precision = getattr(quantizer, "precision", 32)
-            params[precision] = params.get(precision, 0) + int(
-                np.prod(weight.shape.as_list())
-            )
-        for weight in layer.weights:
-            if weight not in layer.quantized_latent_weights:
-                params[32] = params.get(32, 0) + int(np.prod(weight.shape.as_list()))
-        return params
-    return {32: layer.count_params()}
+    @property
+    def count(self):
+        return int(np.prod(self._parameter.shape.as_list()))
 
+    @property
+    def memory(self):
+        return self.bitwidth * self.count
 
-def _sum_params(layer_stats):
-    params = {}
-    for layer_stat in layer_stats:
-        for key, value in layer_stat.items():
-            params[key] = params.get(key, 0) + value
-    return params
+    @property
+    def fp_equivalent_memory(self):
+        return 32 * self.count
 
+    @property
+    def trainable(self):
+        return self._parameter.trainable
 
-def _row_from_stats(stats, summed_stats):
-    return (stats.get(key, 0) for key in sorted(summed_stats))
+    def is_bias(self):
+        return "bias" in self._parameter.name
 
 
-def _get_input_precision(layer):
-    try:
-        return layer.input_quantizer.precision
-    except:
-        return "-"
+@dataclass
+class _operationProfile:
+    n: int
+    precision: int
+    op_type: str
+
 
+class LayerProfile:
+    def __init__(self, layer):
+        self._layer = layer
+        self.parameter_profiles = [
+            _parameterProfile(weight, self._get_bitwidth(weight))
+            for weight in layer.weights
+        ]
+
+        self.op_profiles = []
+
+        if type(layer) in mac_layers:
```

Comment:
Does this do the right thing?

I think if `mac_layers` would be a tuple you can could use:
```python
if isinstance(layer, mac_layers):
```
Not sure if it works with a `list` too.